### PR TITLE
[docs] - Add MdBook for documentation and GHA workflow

### LINKED ISSUE #111: GHA: don't run on markdown file changes
We should update our GHA to skip any documentation-only changes

### PR DESCRIPTION
**Pull request by AshleyDumaine**
This adds [mdBook](https://github.com/rust-lang/mdBook) documentation and a new GHA workflow to build and publish the book to GitHub Pages. 

Also resolves #111

### CODE REVIEW

**Review by rahulait — APPROVED** [review]

File: docs/book/src/developers/development.md [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+kind delete cluster --name tilt
+```
+
+### Deploying a workload cluster
+
+After your kind management cluster is up and running with Tilt, you should be ready to deploy your first cluster.
+
+#### Customizing the cluster deployment
+
+Here is a list of required configuration parameters (the full list is available in `templates/cluster-template.yaml`):
```
[the remark this answers is not in the document]
Comment (AshleyDumaine) [reply docs/book/src/developers/development.md:122]: TIL! Will add 👍

File: docs/book/src/developers/development.md [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+
+This will provision the cluster with the CNI defaulted to [cilium](../topics/addons.md#cilium)
+and the [linode-ccm](../topics/addons.md#ccm) installed.
+
+Next, obtain the provisioned cluster kubeconfig:
+```bash
+clusterctl get kubeconfig $CLUSTER_NAME > $CLUSTER_NAME-kubeconfig
+```
+
+Once the cluster is reachable, create the Kubernetes Secret on the provisioned
```
Comment (eljohnson92) [inline docs/book/src/developers/development.md:181]: just to track the documentation side of this change as well, can we have this secret created as part of the `clusterctl generate` templates?

**Review by rahulait — APPROVED** [review]

File: docs/book/src/developers/development.md [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+export LINODE_MACHINE_TYPE=g6-standard-2
+export CONTROL_PLANE_MACHINE_COUNT=3
+export WORKER_MACHINE_COUNT=3
+
+# Generate SSH key.
+# If you want to provide your own key, skip this step and set LINODE_SSH_KEY to your existing public key.
+SSH_KEY_FILE=.linodeSSHkey
+rm -f "${SSH_KEY_FILE}"
+ssh-keygen -t rsa -b 4096 -f "${SSH_KEY_FILE}" -N '' 1>/dev/null
+export LINODE_SSH_KEY="$(cat "${SSH_KEY_FILE}.pub)"
```
Comment (eljohnson92) [inline docs/book/src/developers/development.md:145]: can we make this a comment to note that it is optional when spinning up a cluster?

File: docs/book/src/developers/development.md [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+# Cluster settings
+export CLUSTER_NAME=capl-cluster
+export KUBERNETES_VERSION=1.29.1
+export LINODE_REGION=us-ord
+
+# Linode settings
+export LINODE_TOKEN=<your linode PAT>
+
+# Machine settings
+export LINODE_OS=linode/ubuntu22.04
```
Comment (eljohnson92) [inline docs/book/src/developers/development.md:134]: can we make this optional since the controller has a default for this

**Review by eljohnson92 — APPROVED** [review]

**Pull request merged**